scripts/require-sync/main.go

```diff
@@ -0,0 +1,193 @@
+				_, err = os.Stat(filepath.Join(path, "sync.yaml"))
+				if os.IsNotExist(err) && *fileFlag {
+					return fmt.Errorf("Error: `sync.yaml` not found for %s", templateName)
+				}
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type KRM struct {
+	ConstraintTemplate *templates.ConstraintTemplate
+	// TODO: Annotations are not queryable in templates.ConstraintTemplate/ObjectMeta
+	Metadata struct {
+		Annotations map[string]string `json:"annotations,omitempty"`
+	} `json:"metadata,omitempty"`
+}
+
+func unMarshall(templatePath string) (*KRM, error) {
+	// read the template into memory
+	var ct *KRM
+	f, err := os.ReadFile(templatePath)
+	if err != nil {
+		return nil, fmt.Errorf("reading file: %w", err)
+	}
+	err = yaml.Unmarshal(f, &ct)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling Annotations: %w", err)
+	}
+	err = yaml.Unmarshal(f, &ct.ConstraintTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling ConstraintTemplate: %w", err)
+	}
+	return ct, nil
+}
+
+func hasAnnotation(annotations map[string]string, annotation string) (bool, error) {
+	if _, exists := annotations[annotation]; exists {
+		return true, nil
+	}
+	return false, nil
+}
+
+func getTemplateName(ct *templates.ConstraintTemplate) (string, error) {
+	return ct.Spec.CRD.Spec.Names.Kind, nil
+}
+
+func isReferential(ct *templates.ConstraintTemplate) (bool, error) {
+	ct.SetName(strings.ToLower(ct.Spec.CRD.Spec.Names.Kind))
+
+	nonRefClient, err := opaClient(false)
+	if err != nil {
+		return false, err
+	}
+
+	// a referential template will fail when added to a client that does not
+	// have the `inventory` field enabled.  Trying to add the template to the
+	// non-referential client thus serves as an indication of it being
+	// referential.
+	_, err = nonRefClient.AddTemplate(context.Background(), ct)
+	if err == nil {
+		// successfully added template to non-referential client.  Template is
+		// non-referential.
+		return false, nil
+	} else if strings.Contains(err.Error(), "check refs failed on module") {
+		// referential data is required.  i.e. we have a referential template
+
+		// do a sanity check that we can add the template to a referential
+		// client
+		refClient, err := opaClient(true)
+		if err != nil {
+			return false, err
+		}
+		if _, err := refClient.AddTemplate(context.Background(), ct); err != nil {
+			return false, fmt.Errorf("adding template to referential client: %w", err)
+		}
+		return true, nil
+	} else {
+		return false, fmt.Errorf("adding template to client: %v", err)
+	}
+}
+
+func opaClient(referential bool) (*opa.Client, error) {
+	externs := local.Externs()
+	if referential {
+		externs = local.Externs("inventory")
+	}
+
+	driver, err := local.New(local.Tracing(false), externs)
+	if err != nil {
+		return nil, fmt.Errorf("creating driver: %w", err)
+	}
+
+	if referential {
+		driver, err = local.New(local.Tracing(false), local.Externs("inventory"))
+	}
+	if err != nil {
```

Comment:
does this need to be moved inside the referential check?

Reply:
Yes, that whole section has been rewritten.